### PR TITLE
Add e2e tests for login page

### PR DESCRIPTION
**Pull request by ronronscelestes**
## WIP

- [x] Adding tests for e2e auth tests
- [x] Finish working on the main setup
- [x] Remove parts of `globalSetup` and `storageState` once the `tar` backup is fully working

### CODE REVIEW

**Comment by ronronscelestes** [comment]
@jhoward1994 @joshuaellis @gu-stav 
PR ready to review
one thing that still bothers us is that to make login tests pass, we had to remove parallelization as we told you
we didn't find a solution to make them work without this configuration - though it's not an ideal configuration
wdyt?

File: e2e/authentication/login.spec.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import { resetDatabaseAndImportDataFromPath } from '../scripts/dts-import';
+
+test.describe('Authentication | Login', () => {
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath({ filePath: './e2e/data/with-admin.tar' });
+    await page.goto('/admin');
+  });
+
+  test.afterEach(async ({ page }) => {
+    await page.request.fetch('/api/database/dump', {
```
Comment (joshuaellis) [inline e2e/authentication/login.spec.js:11]: how come you still have to dump the DB if you're using DTS above?

File: e2e/authentication/login.spec.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      method: 'POST',
+    });
+  });
+
+  test.describe('Successful login', () => {
+    test('A user should be able to log in', async ({ page }) => {
```
Comment (joshuaellis) [inline e2e/authentication/login.spec.js:17]: I think this test case is covered by the other two?

File: e2e/authentication/login.spec.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      await expect(page.getByText('Too many requests, please try again later.')).toBeVisible();
+
+      await page.request.fetch('/api/config/ratelimit/enable', {
+        method: 'POST',
+        data: { value: false },
+      });
+    });
+  });
+
+  test.describe('Validation checks', () => {
```
Comment (joshuaellis) [inline e2e/authentication/login.spec.js:90]: These are very unit testy and we know from experience e2e tests take a long time to run. Could they just be in one giant test?

File: e2e/authentication/login.spec.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      await page.getByLabel('Email*', { exact: true }).fill('test@testing.com');
+      await page.getByLabel('Password*', { exact: true }).fill('wrongPassword');
+
+      await page.getByRole('button', { name: 'Login' }).click();
+
+      await expect(page.getByText('Invalid credentials')).toBeVisible();
+    });
+  });
+
+  test.describe('Other actions than logging in', () => {
```
Comment (joshuaellis) [inline e2e/authentication/login.spec.js:132]: Wondering if we could combine these too 🤔

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Some discussion points...

File: e2e/authentication/login.spec.js (hunk 1 above)
Comment (ronronscelestes) [reply e2e/authentication/login.spec.js:11]: forgot to remove it after we managed to have a working DTS, good catch!

File: e2e/authentication/login.spec.js (hunk 4 above)
Comment (ronronscelestes) [reply e2e/authentication/login.spec.js:104]: done [here](https://github.com/strapi/strapi/pull/15654/commits/165c30139e489e33749bf388f95477bfda6154c3)

File: e2e/authentication/login.spec.js (hunk 3 above)
Comment (ronronscelestes) [reply e2e/authentication/login.spec.js:73]: I agree, done [here](https://github.com/strapi/strapi/pull/15654/commits/165c30139e489e33749bf388f95477bfda6154c3)

**Comment by gu-stav** [comment]
@joshuaellis @jhoward1994 I think this is ready for another review.

**Review by jhoward1994 — APPROVED** [review]
Looks good. Some of the refactoring you've done I should account for in the sign up PR too. Once this is merged I'll catch that branch up with the changes

**Pull request merged**